src/runAll.js

```diff
@@ -0,0 +1,69 @@
+'use strict'
+
+const sgf = require('staged-git-files')
+const Listr = require('listr')
+const stringifyObject = require('stringify-object')
+const getConfig = require('./getConfig')
+const runScript = require('./runScript')
+const generateTasks = require('./generateTasks')
+const resolveGitDir = require('./resolveGitDir')
+
+module.exports = function runAll(packageJson, originalConfig) {
+  const config = getConfig(originalConfig)
+  const gitDir = config.gitDir
+  const verbose = config.verbose
+  const concurrent = config.concurrent
+  const renderer = config.renderer
+  sgf.cwd = resolveGitDir(gitDir)
+
+  if (verbose) {
+    // TODO: Remove config resolution and logging
+    // eslint-disable-next-line
+    console.log(`
+Running lint-staged with the following config:
+${stringifyObject(config)}
+
+`)
+  }
+
+  return new Promise((resolve, reject) => {
+    sgf('ACM', (err, files) => {
+      if (err) {
+        reject(err)
+      }
+
+      /* files is an Object{ filename: String, status: String } */
+      const filenames = files.map(file => file.filename)
+      const tasks = generateTasks(config, filenames).map(task => ({
+        title: `Running tasks for ${task.pattern}`,
+        task: () =>
+          new Listr(runScript(task.commands, task.fileList, packageJson, config), {
+            // In sub-tasks we don't want to run concurrently
+            // and we want to abort on errors
+            concurrent: false,
+            exitOnError: true
+          }),
+        skip: () => {
+          if (task.fileList.length === 0) {
+            return `No staged files match ${task.pattern}`
+          }
+          return false
+        }
+      }))
+
+      if (tasks.length) {
```

Comment:
What happens if `tasks.length` is 0? The promise will not resolve right?

Reply:
Good catch! I'll add a resolution for this case. Need more tests!